src/routes/product-list/index.tsx

```diff
@@ -11,36 +11,47 @@ import { useAtom } from "jotai"
 import { SiteKey } from "../../types/site"
 import { FullPageLoader } from "../../components/Loader"
 import { useSiteChanged } from "../../utils/use-site-changed"
+import { getCategoryList } from "../../utils/query-category"
 
 interface Props {
   categoryId?: string
 }
 
 export const ProductAnalyticsPage = (props: Props) => {
   const [site] = useAtom(siteAtom)
-  const categoryId = props.categoryId && parseInt(props.categoryId, 10)
+
+  const categoryId = props.categoryId ? parseInt(props.categoryId, 10) : null
```

Comment:
why do we need to cast from undefined to null when it's not passed?

Reply:
This is just to make the type a bit nicer as the type used to include empty string in the union.

![CleanShot 2568-01-09 at 18 21 25@2x](https://github.com/user-attachments/assets/c8c446ef-c953-4cfd-a007-9fce75591882)